api/src/oscc.c

```diff
@@ -418,23 +496,38 @@ void oscc_update_status( )
             }
             else
             {
-                if ( obd_frame_callback != NULL )
+                if ( obd_frame_callback != NULL && vehicle_can_socket < 0 )
                 {
                     obd_frame_callback( &rx_frame );
                 }
             }
 
-            ret = read( can_socket, &rx_frame, CAN_MTU );
+            ret = read( oscc_can_socket, &rx_frame, CAN_MTU );
         }
     }
+
+    if ( vehicle_can_socket != -1 )
+    {
+      int veh_ret = read( vehicle_can_socket, &rx_frame, CAN_MTU );
+
+      while( veh_ret > 0 )
+      {
+          if ( obd_frame_callback != NULL )
```

Comment:
If the OBD callback is not set then there's no point in reading from the vehicle CAN bus, so I think we can move this conditional outside of the while loop. Something about not reading this socket worries me a bit though - I have some weird concern that not reading from this buffer might cause something to block but I don't know what would cause this.

As-is I think this is slightly less efficient but it might be easier to leave this as-is. Your call on if you want to dig into optimizing this.